CMakeLists.txt

```diff
@@ -24,13 +25,26 @@ cmake_dependent_option(LIBUV_BUILD_BENCH
   "Build the benchmarks when building unit tests and we are the root project" ON
   "LIBUV_BUILD_TESTS" OFF)
 
-if(MSVC)
-  list(APPEND uv_cflags /W4)
-elseif(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
-  list(APPEND uv_cflags -Wall -Wextra -Wstrict-prototypes)
-  list(APPEND uv_cflags -Wno-unused-parameter)
+# Unfortunately, MSVC and clang-cl both support -Wall, but we don't actually
+# want that, as its equivalent to -Weverything :/
+if (MSVC OR CMAKE_C_SIMULATED_ID STREQUAL "MSVC")
```

Comment:
Google turns up zero matches for `CMAKE_C_SIMULATED_ID`, that's not something you see everyday anymore. ^^

Can you elaborate little on what it does and where it comes from? (That CheckCCompilerFlag module?)

Reply:
Ahh this appears to actually be a typo. The correct string is `CMAKE_C_SIMULATE_ID`. You can read more about it [here][1]

[1]: https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_SIMULATE_ID.html

Reply:
CMake, of course, makes it very difficult to find things sometimes 😅. The `CMAKE_C_SIMULATE_ID` is a more recent variable that is set. The reason we want to check for it is if someone is using clang-cl, which is the clang driver with an msvc style frontend. If someone uses an older version of cmake it won't work with clang-cl to begin with (for reasons I'd rather not go into 🙄), but this allows us to be forward compatible.

Reply:
Ah, very nice. Thanks for explaining!